packages/react-dom/src/events/accumulateTwoPhaseListeners.js

```diff
@@ -127,6 +135,30 @@ export default function accumulateTwoPhaseListeners(
             dispatchInstances.push(node);
           }
         }
+      } else if (enableScopeAPI && node.tag === ScopeComponent) {
```

Comment:
I'm assuming this logic is from flare's implementation?

Reply:
Yep